src/__tests__/linter.test.ts

```diff
@@ -1251,4 +1253,61 @@ responses:: !!foo
       ]);
     });
   });
+
+  test.each(['1', 'null', '', 'false'])('given %s input, should report nothing', async input => {
```

Comment:
In the Ruby/RSpec world we'd try to test intent more than the result. Instead of saying "assuming this whole thing runs like this, we will see these results, or those other results", we'd say "we expect this internal function to be called, or not called". I know its possible with mosts in jest, so im just wondering if we are intentionally avoiding it? 

Being able to test that `this.resolver.resolve` is called (or not) would make is faaaar easier to understand what this test is actually trying to prove IMO.

Reply:
>Being able to test that this.resolver.resolve is called (or not) would make is faaaar easier to understand what this test is actually trying to prove IMO.

With regards to _this_ test, I don't see it as trying to assess that the resolver is leveraged or not (which is somehow an implementation detail), but rather that such input doesn't lead to a result such as `"'undefined' does not exist"` (which was the initial issue). Having this covered in such a way proves that the bug is solved and won't regress.

>we'd say "we expect this internal function to be called, or not called". I know its possible with mosts in jest, so im just wondering if we are intentionally avoiding it?

In my experience, those kind of tests tend to become very brittle in the long run. Every internal refactoring may lead those tests to fail (for the wrong reasons).